assets/js/payment-method-extensions/payment-methods/stripe/payment-request/payment-request-express.js

```diff
@@ -0,0 +1,380 @@
+/**
+ * Internal dependencies
+ */
+import {
+	DEFAULT_STRIPE_EVENT_HANDLERS,
+	PAYMENT_METHOD_NAME,
+} from './constants';
+import {
+	getStripeServerData,
+	getPaymentRequest,
+	updatePaymentRequest,
+	canDoPaymentRequest,
+	getTotalPaymentItem,
+	getBillingData,
+	getPaymentMethodData,
+	getShippingData,
+	normalizeShippingAddressForCheckout,
+	normalizeShippingOptions,
+	normalizeLineItems,
+	normalizeShippingOptionSelectionsForCheckout,
+} from '../stripe-utils';
+
+/**
+ * External dependencies
+ */
+import { useRef, useState, useEffect } from '@wordpress/element';
+import {
+	Elements,
+	PaymentRequestButtonElement,
+	useStripe,
+} from '@stripe/react-stripe-js';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * @typedef {import('../stripe-utils/type-defs').Stripe} Stripe
+ * @typedef {import('../stripe-utils/type-defs').StripePaymentRequest} StripePaymentRequest
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').RegisteredPaymentMethodProps} RegisteredPaymentMethodProps
+ */
+
+/**
+ * @typedef {Object} WithStripe
+ *
+ * @property {Stripe} [stripe] Stripe api (might not be present)
+ */
+
+/**
+ * @typedef {RegisteredPaymentMethodProps & WithStripe} StripeRegisteredPaymentMethodProps
+ */
+
+/**
+ * PaymentRequestExpressComponent
+ *
+ * @param {StripeRegisteredPaymentMethodProps} props Incoming props
+ */
+const PaymentRequestExpressComponent = ( {
+	shippingData,
+	billing,
+	eventRegistration,
+	onSubmit,
+	activePaymentMethod,
+	setActivePaymentMethod,
+	setExpressPaymentError,
+} ) => {
+	/**
+	 * @type {[ StripePaymentRequest|null, function( StripePaymentRequest ):StripePaymentRequest|null]}
+	 */
+	// @ts-ignore
+	const [ paymentRequest, setPaymentRequest ] = useState( null );
+	const stripe = useStripe();
+	const [ canMakePayment, setCanMakePayment ] = useState( false );
+	const [ paymentRequestType, setPaymentRequestType ] = useState( '' );
+	const [ isProcessing, setIsProcessing ] = useState( false );
+	const [ isFinished, setIsFinished ] = useState( false );
```

Comment:
This payment method needs to track it's internal state of processing or finished. It can't use `paymentStatus` because that has potentially different meanings in the checkout context. Also payment status is for whatever the active payment method is and we still need to initialize some things in this payment method regardless of the payment status. 

Also, event observers _cannot update context state_ but only local state, so that has a huge influence on this approach.

This potentially can be revisited (and maybe use a combination of `activePaymentMethod` along with `paymentStatus`) but for now, these states make things simpler to reason about.